File: week6/Baekjoon28290/sjYeo/Baekjoon28290.cpp
#include<iostream>
#include<string>
#include<cstring>

using namespace std;

// size=6
static char* typings[] = {
    "fdsajkl;", // 1 
    "jkl;fdsa", // 1
    "asdf;lkj", // 2
    ";lkjasdf", // 2
    "asdfjkl;", // 3
    ";lkjfdsa"  // 4
};

int main(void){
    char u_in[9];
    //iostream, 마지막 NULL문자를 제외한 n-1개의 문자를 읽어와 str 배열에 저장
    cin.getline(u_in, sizeof(u_in));
    int i;
    for(i=0; i<6; i++){
        if(strcmp(typings[i], u_in)==0){ //문자열이 같으면
            switch(i){
                case 0:
                case 1: {printf("in-out\n"); return 0;}
                case 2: 
                case 3: {printf("out-in\n"); return 0;}
                case 4: {printf("stairs\n"); return 0;}
                case 5: {printf("reverse\n"); return 0;}
            }
        }
    }
    printf("molu\n");
    return 0;
}

File: week6/Baekjoon28290/gihwanJang/Baekjoon28290.java
import java.io.*;
import java.util.*;

public class Baekjoon28290 {
    Map<String, String> map;

    public Baekjoon28290() {
        map = new HashMap<>(5);
        map.put("fdsajkl;", "in-out");
        map.put("jkl;fdsa", "in-out");
        map.put("asdf;lkj", "out-in");
        map.put(";lkjasdf", "out-in");
        map.put("asdfjkl;", "stairs");
        map.put(";lkjfdsa", "reverse");
    }

    public String solve(String s) {
        return map.containsKey(s) ? map.get(s) : "molu";
    }

    public static void main(String[] args) throws IOException {
        Baekjoon28290 problem = new Baekjoon28290();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        System.out.println(problem.solve(br.readLine()));
    }
}